streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -248,12 +248,11 @@ public void handleAssignment(final Map<TaskId, Set<TopicPartition>> activeTasks,
         );
 
         final LinkedHashMap<TaskId, RuntimeException> taskCloseExceptions = new LinkedHashMap<>();
-
         final Map<TaskId, Set<TopicPartition>> activeTasksToCreate = new HashMap<>(activeTasks);
         final Map<TaskId, Set<TopicPartition>> standbyTasksToCreate = new HashMap<>(standbyTasks);
-        final List<Task> tasksToClose = new LinkedList<>();
-        final Set<Task> tasksToRecycle = new HashSet<>();
-        final Set<Task> dirtyTasks = new HashSet<>();
+        final List<Task> tasksToRecycle = new LinkedList<>();
+        final List<Task> tasksToCloseClean = new LinkedList<>();
+        final List<Task> tasksToCloseDirty = new LinkedList<>();
```

Comment:
Why do we maintain `List` instead of `Set` ? It seem more natural to me to have a `Set` (ie, should we instead switch `tasksToRecyle` to use a `Set`, too)?

Reply:
Two reasons: 1) in unit tests we are validating some scenarios with ordering assumptions, 2) I'd like to have log4j entries to have ordering as well.

Reply:
Can we use an ordered set then? Just to make sure we can't end up with a task appearing more than once in the same list/set

Reply:
Sounds good, I can try that.